### DISCUSSION TITLE
connector的对this的使用为什么不改成shared_from_this()?

### DISCUSSION THREAD

**Discussion by zywww** [chenshuo/muduo#591] [General]
connector.cc 里面的那些 fixme 注释的 this 为什么不用 shared_from_this() 代替?
如果这样修改的话
TcpClient 里面的这行代码也可以去掉了把?
loop_->runAfter(1, std::bind(&detail::removeConnector, connector_));

**Comment by chenshuo** [comment]
你可以把你的想法用代码具体实现出来，我们再来谈。
